comma/hypothesis.py

```diff
@@ -26,63 +30,312 @@ class Hypothesis:
+
+        min_date = min(all_dates)
+        max_date = max(all_dates)
+
+        if start < min_date or end > max_date:
+            raise ValueError(
+                f"time_period ({time_period[0]} - {time_period[1]}) "
+                f"is outside available dates that go from "
+                f"({min_date} to {max_date})"
+            )
+
+        # Filtering the dates now that we know they're within the range
+        filtered_paths = [
+            file
+            for file in file_list
+            if start
+            <= datetime.strptime(pattern.search(file).group(1), "%Y-%m-%d")
+            <= end
+        ]
+
+        return filtered_paths
+
+    @staticmethod
+    def compute_time_period(start: str, steps: int, date_format: str) -> tuple:
+        """
+        Compute time period based on a starting date and number of steps
+
+        Args:
+            start (str): Start date
+            steps (int): Number of steps
+            date_format (str): Format of the date string
+
+        Returns:
+            tuple: A tuple containing the start and end date
+        """
+        start_date = datetime.strptime(start, date_format)
+        end_date = start_date + timedelta(days=steps)
+        return (start_date.strftime(date_format), end_date.strftime(date_format))
+
+    @classmethod
+    def get_covid_data(
+        cls, time_period: Tuple[str, str], location: str
+    ) -> pd.DataFrame:
+        """
+        Download and filter COVID-19 test data from the RIVM website.
+
+        Args:
+        time_period (tuple): Start and end date ('YYYY-MM-DD', 'YYYY-MM-DD').
+        location (str): Security region name. This is the name of the city.
+
+        Returns:
+        df_filtered (pandas.DataFrame): Filtered data.
+        """
+
+        dates = cls.get_file_paths(cls.RIVM_URL)
+        filtered_dates = cls.filter_dates(dates, time_period)
+
+        df_gzip = []
+        message = "Downloading COVID-19 data from RIVM"
+        for date in tqdm(filtered_dates, desc=message):
+            full_url = cls.RIVM_URL + date.split("/")[-1]
+            df = pd.read_csv(
+                full_url, compression="gzip", header=0, sep=",", quotechar='"'
+            )
+            if not isinstance(df, pd.DataFrame):
+                raise ValueError(
+                    f"Data retrieved from {cls.RIVM_URL}"
+                    f" is not a DataFrame but a {type(df)}"
+                )
+            df_gzip.append(df)
+
+        df_tests = pd.concat(df_gzip, ignore_index=True)
+
+        df_tests["Date_of_statistics"] = pd.to_datetime(df_tests["Date_of_statistics"])
+
+        mask = (
+            (df_tests["Date_of_statistics"] >= time_period[0])
+            & (df_tests["Date_of_statistics"] <= time_period[1])
+            & (df_tests["Security_region_name"] == location)
+        )
+        df_filtered = df_tests.loc[mask].reset_index(drop=True)
+
+        return df_filtered
```

Comment:
Personally, I would try to make this a bit more concise:

```suggestion
        return df_tests.loc[df_tests["Date_of_statistics"] >= time_period[0]
            & df_tests["Date_of_statistics"] <= time_period[1]
            & df_tests["Security_region_name"] == location]
        .reset_index(drop=True)
```
Just to avoid the intermediate `mask` and `df_filtered` variables (and `black` should format this a bit more nicely).